### PR TITLE
Added support for 16 bit bus for I2CMasterBus

### PR DESCRIPTION
**Pull request by Cristian243342**
### Pull Request Overview

This pull request adds support for 16 bit bus for I2CMasterBus.

### Testing Strategy

This pull request is untested.

### TODO or Help Wanted

This pull request still needs testing.

### Documentation Updated

- [ ] No updates are required.

### Formatting

- [ ] Ran `make prepush`.

### CODE REVIEW

File: capsules/extra/src/bus.rs [hunk 1]
```diff
@@ -301,7 +301,21 @@ impl<'a, I: I2CDevice> Bus<'a> for I2CMasterBus<'a, I> {
                         }
                     }
                 }),
-
+            BusWidth::Bits16LE => self
```
Comment (alexandruradovici) [inline capsules/extra/src/bus.rs:304]: Can we add all the variants here? The code should be straight forward.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

File: capsules/extra/src/bus.rs [hunk 2]
```diff
@@ -301,7 +301,21 @@ impl<'a, I: I2CDevice> Bus<'a> for I2CMasterBus<'a, I> {
...
+                .addr_buffer
+                .take()
+                .map_or(Err(ErrorCode::NOMEM), |buffer| {
+                    buffer[0] = 0x00;
```
Comment (alexandruradovici) [inline capsules/extra/src/bus.rs:308]:
```suggestion
                    buffer[0] = addr & 0xff;
```

File: capsules/extra/src/bus.rs [hunk 3]
```diff
@@ -301,7 +301,21 @@ impl<'a, I: I2CDevice> Bus<'a> for I2CMasterBus<'a, I> {
...
+                    buffer[1] = addr as u8;
```
Comment (alexandruradovici) [inline capsules/extra/src/bus.rs:309]:
```suggestion
                    buffer[1] = (addr >> 8) as u8;
```

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

**Comment by alistair23** [comment]
Why add this if it's untested? If it is not being used we don't need to support it

**Comment by lschuermann** [comment]
> Why add this if it's untested? If it is not being used we don't need to support it

This is used and @alexandruradovici explicitly asked for it to be split out in #3811.

@Cristian243342 Do you have any updates on this?

**Comment by lschuermann** [comment]
@alexandruradovici, do you know how to best proceed here?

**Comment by alexandruradovici** [comment]
This is not yet done, I'll talk to my student and get back on this.

**Comment by lschuermann** [comment]
@Cristian243342 @alexandruradovici  friendly ping on this.

**Comment by lschuermann** [comment]
Closing due to staleness, feel free to re-open if there's progress.

**Pull request closed without merging**